cf-agent/files_editline.c

```diff
@@ -1470,34 +1479,76 @@ static int MatchPolicy(EvalContext *ctx, const char *camel, const char *haystack
                             while (isspace((int)*(sp + 1)))
                             {
                                 sp++;
+                                required_size++;
                             }
 
-                            strcat(spto, "\\s+");
-                            spto += 3;
+                            required_size += 3;
+                            strlcpy(toadd, "\\s+", sizeof(toadd));
                         }
                         else
                         {
-                            *spto++ = *sp;
+                            required_size++;
+                            toadd[0] = *sp;
+                            toadd[1] = '\0';
                         }
                     }
                     else
                     {
-                        *spto++ = *sp;
+                        required_size++;
+                        toadd[0] = *sp;
+                        toadd[1] = '\0';
+                    }
+
+                    if (required_size > work_size)
+                    {
+                        work_size = required_size + 12;
```

Comment:
I take it that this ``+ 12``, and the ``+ 6`` further up is simply heuristics, expecting some characters to replaced, but not too many, and then expanding if necessary?